oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/OAuth2TokenDescriptorBuilder.java

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.auth.oauth2;
+
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.ACTIVE;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.AUDIENCE;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.CLIENT_ID;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.EXPIRES_AT;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.ISSUED_AT;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.ISSUER;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.JSON;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.JWT_ID;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.NOT_BEFORE;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.SCOPE;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.SCOPE_SEPARATOR;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.SUBJECT;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.TOKEN_TYPE;
+import static com.linecorp.armeria.common.auth.oauth2.OAuth2TokenDescriptor.USER_NAME;
+import static java.util.Objects.requireNonNull;
+
+import java.time.Instant;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * Builds an instance of {@link OAuth2TokenDescriptor}.
+ */
+public class OAuth2TokenDescriptorBuilder {
+
+    private static final TypeReference<LinkedHashMap<String, String>> MAP_TYPE =
+            new TypeReference<LinkedHashMap<String, String>>() {};
+
+    static OAuth2TokenDescriptor of(String rawResponse) {
```

Comment:
Should this method be moved to `OAuth2TokenDescriptor` and renamed to `builder(String)`?

Reply:
it's already there:
```
    public static OAuth2TokenDescriptor of(String rawResponse) {
        return OAuth2TokenDescriptorBuilder.of(rawResponse);
    }

```

Reply:
I actually meant inlining and removing this method.